packages/backend/src/build-disk-image.ts

```diff
@@ -112,9 +112,9 @@ export async function buildDiskImage(build: BootcBuildInfo, history: History, ov
   const telemetryData: Record<string, unknown> = {};
   telemetryData.build = build;
 
-  // "Returning" withProgress allows PD to handle the task in the background with building.
+  // Kick off task using withProgress to build in the background
   let errorMessage: string;
-  return extensionApi.window
+  extensionApi.window
```

Comment:
IMHO it does not make much sense to have a function with the following signature (promise)

https://github.com/podman-desktop/extension-bootc/blob/8fc7abafe8e4b5d5c29a439c43f41759a16be85a/packages/backend/src/build-disk-image.ts#L60

and having a dangling promise not awaited at the end. If the caller do not to bother to wait, this is the caller issue, not the method implementation. 

The signature of the method is misleading then. In ai-lab we solved it by having two method:

```
// request => do not wait for the task to be finishes, only start it (do not return a promise)
requestCreateInferenceServer(): string;

// awaiting this promise will wait for full completion (return a promise)
createInferenceServer(): Promise<string>;
```

Reply:
Understood, but launching a build requires validation, checking prereqs on Podman, updating the build history, then launching the 2+ minute task to actually do the build. We don't want a function that waits for the full build to complete, but even those initial things require awaits and might take a couple seconds.

IMHO having the promise is accurate and still allows you to decide between awaiting to confirm that the build actually kicked off vs backgrounding it. Would a comment or renaming help with your concern instead?

Reply:
@axel7083 ok? This PR is just 'fixing the current intended behaviour' so even if you think this is wrong I'd prefer to fix the bug and take a followup. :)